core/lib/zksync_core/src/native_token_fetcher/mod.rs

```diff
@@ -93,11 +106,25 @@ impl ConversionRateFetcher for NativeTokenFetcher {
             .await
         {
             Ok(response) => {
-                let conversion_rate = response.json::<u64>().await.context(
+                let conversion_rate = response.json::<BigDecimal>().await.context(
                     "Unable to parse the response of the native token conversion rate server",
                 )?;
-                self.latest_to_eth_conversion_rate
-                    .store(conversion_rate, std::sync::atomic::Ordering::Relaxed);
+                match self.latest_to_eth_conversion_rate.lock() {
+                    Ok(mut lock) => {
```

Comment:
Same as [here](https://github.com/lambdaclass/zksync-era/pull/204/files#r1521443711).

(this comment should be ported to [this PR](https://github.com/matter-labs/zksync-era/pull/1324) once merged).